packages/datetime/src/components/date-input/dateInput.tsx

```diff
@@ -20,7 +20,7 @@
  * package instead.
  */
 
-/* eslint-disable @typescript-eslint/no-deprecated, @blueprintjs/no-deprecated-components, react-hooks/exhaustive-deps */
+/* eslint-disable @typescript-eslint/no-deprecated, react-hooks/rules-of-hooks, react-hooks/exhaustive-deps */
```

Comment:
Had to disable `rules-of-hooks` in some components because some have names starting with underscore, and the linting rule enforces hook names to begin with a capital letter.

Reply:
Doesn't need to be this PR, but we should follow up on these components to rename them to something more compliant with this rule. Having it be disabled is a bit sub-optimal given how important it is.

Reply:
Opening #7259 to address this separately

Reply:
@ggdouglas I confirm that I can drop the last commit with your PR :)